src/components/Theme.svelte

```diff
@@ -0,0 +1,63 @@
+<script lang="ts">
+    import { writable } from "svelte/store";
+
+    import { IS_SERVER } from "../scripts/constants";
+
+    const options = ["Light", "Dark"] as const;
+
+    const initialValue = IS_SERVER
+        ? "Dark"
+        : (() => {
+            const stored = localStorage.theme;
+            if (stored && options.includes(stored)) return stored;
+
+            if (window.matchMedia && window.matchMedia('(prefers-color-scheme: light').matches)
+                return "Light";
+            else return "Dark";
+          })();
+    
+    const selected = writable(initialValue);
+    let first = true;
+    // Note - Theme switcher does not work on dev mode
+    if (!IS_SERVER) {
+        selected.subscribe(v => {
+            localStorage.theme = v;
+            v = v.toLowerCase();
+            const prev = (v === "light") ? "dark" : "light";
+            if (first) {
+                document.body.classList.add(v);
+                first = false;
+            }
+            else document.body.classList.replace(prev, v);
+        });
+    }
+
+    function themeSwitch() {
+        selected.update(x => x = (x === "Light") ? "Dark" : "Light");
+    }
+</script>
+
+<!-- Probably not the best way to go about this tbh -->
+<button on:click={themeSwitch} title="Switch to Dark">
+    {#if $selected === "Light"}
+        <slot name="light"/>
+    {:else if $selected === "Dark"}
+        <slot name="dark"/>
+    {/if}
+</button>
+
+<style>
+    button {
+        all: unset;
```

Comment:
this removes the focus outline (see screenshot) which is important for people using keyboard navigation
![image](https://user-images.githubusercontent.com/45497981/231902956-ccf9b21d-aa99-4be2-9a5e-2e758094d2cf.png)

Please add it back on button:focus